spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/JpaRepositoryRestResourceIT.java

```diff
@@ -0,0 +1,154 @@
+import java.util.List;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.ts.spring.data.AbstractDbIT;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class JpaRepositoryRestResourceIT extends AbstractDbIT {
+
+    @Test
+    void allRepositoryMethodsValidRequests() {
+        //GET - List all articles (should have all 4 articles the database has initially)
+        app.given()
+                .accept("application/json")
+                .when().get("/article-jpa")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(
+                        containsString("Aeneid"),
+                        containsString("Beach House"),
+                        containsString("Cadillac Desert"),
+                        containsString("Dagon and Other Macabre Tales"));
+
+        //GET - List all articles in hal+json format
+        String articlesList = app.given()
+                .accept("application/hal+json")
+                .when().get("/article-jpa")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract().response().jsonPath().getString("_embedded.article-jpa.name");
+        assertEquals("[Aeneid, Beach House, Cadillac Desert, Dagon and Other Macabre Tales]", articlesList);
+
+        //POST - Create a new Article
+        app.given()
+                .contentType("application/json")
+                .accept("application/json")
+                .body("{\"name\": \"Early Asimov\", \"author\": \"Isaac Asimov\"}")
+                .when().post("/article-jpa")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .body(containsString("Early Asimov"))
+                .body("id", notNullValue())
+                .extract().body().jsonPath().getString("id");
+
+        //PUT - Update a new Article
+        app.given()
+                .contentType("application/json")
+                .accept("application/json")
+                .body("{\"name\": \"Early Asimov 2nd Edition\", \"author\": \"Isaac Asimov\"}")
+                .when().put("/article-jpa/5")
+                .then()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+
+        //GET{id} - Find new article by id
+        app.given()
+                .when().get("/article-jpa/5")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(
+                        containsString("Early Asimov 2nd Edition"));
+
+        //DELETE - Try to delete a article via HTTP (method not allowed)
+        app.given()
+                .when().delete("/article-jpa/5")
+                .then()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+
+        //Test repository pagination
+        app.given()
+                .accept("application/json")
+                .queryParam("size", "2")
+                .queryParam("page", "0")
```

Comment:
I suggest to check any other page, but first, to verify, that pagination indeed works as expected. But it's not a requirement.

Reply:
Pagination test expanded.